ideas/xx-message-ordering.md

```diff
@@ -0,0 +1,42 @@
+## Preamble
+
+    Idea: #??-message-ordering
+    Title: Message ordering
+    Status: Draft
+    Created: 2018-03-29
+
+
+## Summary
+
+Addresses point 2 and 3 of https://github.com/status-im/ideas/issues/99 .
+
+Define, document and implement message ordering for 1-to-1, group and public chats so that we can give the best UX experience to the end user.
+
+## Swarm Participants
+
+- Lead & PM: @cammellos
+- Testing & Evaluation: 
+- UX/Design: 
+
+## Product Overview
+
+Because status is a fully decentralized app, message ordering is non-trivial and needs careful thinking in terms of UX, as some constrainsts needs to be taken into consideration.
+
+Currently message ordering in the chat is by the time the message has been received which results in out-of-order messages and potentially different order on different devices.
+
+The primary aim of this swarm is to define which message ordering gives users the best UX experience and implement the changes necessary (if any).
+
+**Goals**
+- Define, document and implement ordering of messages for 1-to-1/group/public chats
+
```

Comment:
What do you think a first iteration looks like?

Reply:
I think the first iteration would produce basically everything described, docs + ordering + tests, at the moment it seems like it's manageable.

Other slices would be a bit artificial I feel, as test + docs + ordering feel like something should really come together.

Out of scope would be placeholders and potentially displaying the timestamps of messages.

Those could either go on another swarm (definitely placeholders), or added as a second iteration, but I would want to have a word with UX before committing, just to understand what their plans/ideas are.